pages/api/session.js

```diff
@@ -1,83 +1,65 @@
 import People from '@klaudsol/commons/models/People';
 import { withSession } from '@klaudsol/commons/lib/Session';
-import { defaultErrorHandler } from '@klaudsol/commons/lib/ErrorHandler';
-import { setCORSHeaders } from "@klaudsol/commons/lib/API";
 import { OK, UNPROCESSABLE_ENTITY } from '@klaudsol/commons/lib/HttpStatuses';
 import UnauthorizedError from '@klaudsol/commons/errors/UnauthorizedError';
 import Session from '@klaudsol/commons/models/Session';
-import { assertUserIsLoggedIn } from '@klaudsol/commons/lib/Permissions';
-import EntityType from '@/backend/models/core/EntityType';
-import { setCookie } from 'cookies-next';
+import { handleRequests } from '@klaudsol/commons/lib/API';
 import { generateToken } from '@klaudsol/commons/lib/JWT';
+import { canLogInToCms } from '@/lib/Constants';
 
-export default withSession(handler);
+const FRONTEND_URL = process.env.KS_FRONTEND_URL ?? process.env.FRONTEND_URL
 
+export default withSession(handleRequests({ post, del }));
 
-async function handler(req, res) {
-  // i hate cors
-  setCORSHeaders({ response: res, url: process.env.KS_FRONTEND_URL ?? process.env.FRONTEND_URL });
+async function post(req, res) {
+    const { email = null, password = null } = req.body;
 
-  try {
-    switch(req.method) {
-      case "POST":
-        return await login(req, res);
-      case "DELETE":
-        return await logout(req, res); 
-      case "OPTIONS":
-        return res.status(OK).json({});
-      default:
-        throw new Error(`Unsupported method: ${req.method}`);
+    if (!email || !password) {
+        res.status(UNPROCESSABLE_ENTITY).json({ message: "Please enter your username/password." });
+        return
     }
-  } catch (error) {
-    await defaultErrorHandler(error, req, res);
-  }
-}
 
-async function login (req, res) {   
-  try {
+    const { session_token, user: { firstName, lastName, capabilities, defaultEntityType, forcePasswordChange } } = await People.login(email, password);
 
-    const {email=null, password=null} = req.body; 
-  
-    if (!email || !password) {
-      res.status(UNPROCESSABLE_ENTITY).json({message: "Please enter your username/password."});   
-      return
-      }
+    const token = generateToken({ firstName, lastName, email, capabilities, sessionToken: session_token });
+    const response = { message: 'Sucessfully logged in!' };
+
+    const { origin } = req.headers;
+    const isFromCMS = origin !== FRONTEND_URL;
```

Comment:
I'm not 100% onboard with having security depend on URL's and environment variables. 🤔